R/run-nmtran.R

```diff
@@ -0,0 +1,375 @@
+#'
+#' @keywords internal
+nmtran_setup <- function(
+    .mod,
+    .bbi_args = NULL,
+    .nonmem_version = .bbi_args[["nm_version"]],
+    .config_path = NULL
+){
+  bbi_yaml_path <- get_bbi_yaml_path(.mod, .config_path = .config_path)
+  bbi_yaml <- yaml::read_yaml(bbi_yaml_path)
+
+  # Combine NONMEM submission args
+  #  - The main ones of interest are prdefault, tprdefault, and maxlim, which
+  #    impact the evaluation of `NM-TRAN`
+  #  - Priority: .bbi_args > model yaml > bbi.yaml
+  cmd_args <- parse_nmtran_args(
+    .mod, .bbi_args = .bbi_args, nmfe_options = bbi_yaml$nmfe_options
+  )
+
+  # Check nonmem version
+  nm_config <- bbi_yaml$nonmem
+  if(is.null(nm_config)){
+    cli::cli_abort(
+      c(
+        "Did not find required NONMEM information in configuration file: {.path {bbi_yaml_path}}",
+        "i" = "Please see {.fun bbr::bbi_init} for additional details"
+      )
+    )
+  }
+  if (!is.null(.nonmem_version)) {
+    # check for valid version
+    if (!(.nonmem_version %in% names(nm_config))) {
+      cli::cli_abort(
+        c(
+          "Must specify a valid {.var nm_version} for {.fun bbr::run_nmtran}.",
+          "i" = "{.path {bbi_yaml_path}} contains the following options:",
+          "{names(nm_config)}",
+          "*" = "e.g., {.code run_nmtran(mod, .bbi_args = list(nm_version = '{names(nm_config)[1]}'))}"
+        )
+      )
+    }
+    default_nm <- nm_config[[.nonmem_version]]
+  }else{
+    # Look for default nonmem installation
+    default_nm <- purrr::keep(nm_config, function(nm_ver){
+      !is.null(nm_ver$default) && isTRUE(nm_ver$default)
+    })
+
+    if(length(default_nm) > 1){
+      nm_vers <- paste(names(default_nm), collapse = ", ")
+      cli::cli_abort(
+        c(
+          "Found multiple default NONMEM versions ({nm_vers}) at {.path {bbi_yaml_path}}",
+          "i" = "Please ensure only one version is set to the default"
+        )
+      )
+    }else if(length(default_nm) == 1){
+      default_nm <- default_nm[[1]]
+    }else{
+      # If no default, use the last one (likely higher version)
+      default_nm <- nm_config[[length(nm_config)]]
+    }
+  }
+
+  # Set NM-TRAN executable path
+  nm_path <- default_nm$home
+  nmtran_exe <- file.path(nm_path, "tr", "NMTRAN.exe")
+
+  # Check if nmtran_presort exists
+  nmtran_presort_exe <- file.path(nm_path, "util", "nmtran_presort")
+  run_presort <- unname(fs::file_exists(nmtran_presort_exe))
+
+  return(
+    list(
+      nmtran_exe = nmtran_exe,
+      nonmem_version = basename(nm_path),
+      cmd_args = cmd_args,
+      nmtran_presort_exe = if(run_presort) nmtran_presort_exe else NULL
+    )
+  )
+}
+
+#' Execute `NM-TRAN` in a given directory
+#'
+#' Execute `NM-TRAN` in a given directory. Also runs `nmtran_presort` if an
+#' executable is found.
+#'
+#' @param nmtran_exe Path to `NM-TRAN` executable.
+#' @param mod_path Path of a model to evaluate. Should be relative to `dir`. Nested
+#'  directories not supported.
```

Comment:
I don't know what the "nested directories" bit of this `mod_path` description is referring to. Something related to bootstrap or simulation models?

Reply:
I can touch this up a bit. That description doesnt address what I was trying to warn against, which is that the model cannot be located at `dir/some_dir/1.ctl`. It must be within `dir` at the top level. Though as I mention that description is wrong since `mod_path` can be any path, and points to where the original model is, rather than the one being executed by `NM-TRAN`